app/models/pglogical_subscription.rb

```diff
@@ -0,0 +1,116 @@
+# This model wraps a pglogical stored proc (pglogical.show_subscription_status)
+# This is exposed to us through the PostgreSQLAdapter#pglogical object's #subscriptions method
+# This model then exposes select values returned from that method
+class PglogicalSubscription < ActsAsArModel
+  set_columns_hash(
+    :id              => :string,
+    :status          => :string,
+    :dbname          => :string,
+    :host            => :string,
+    :user            => :string,
+    :password        => :string,
+    :port            => :integer,
+    :provider_region => :integer
+  )
+
+  def self.find(*args)
+    subs = pglogical.enabled? ? pglogical.subscriptions : []
+    case args.first
+    when :all then subs.collect { |s| new(subscription_to_columns(s)) }
+    when :first, :last then subs.presence && new(subscription_to_columns(subs.send(args.first)))
+    end
+  end
+
+  def save!
+    raise "Cannot update an existing subscription" if id
+    create_node.check if !pglogical.enabled? || self.class.count == 0
+    id = "subscription_#{host.gsub(/\.|-/, "_")}"
+    pglogical.subscription_create(id, dsn, ['miq'], false).check
```

Comment:
And to @gtanzillo 's point, this should reference `MiqPglogical::REPLICATION_SET_NAME` when #7111 is merged